### DISCUSSION TITLE
How to configure DALL-E plugin?

### DISCUSSION THREAD

**Discussion by Nihisil** [danny-avila/LibreChat#1755] [Help Wanted]
I've set the `DALLE_API_KEY` variable in my configuration, but when I'm trying to install the plugin, it asks me to enter the key again. What else do I need to set up to stop this from happening?

<img width="1508" alt="image" src="https://github.com/danny-avila/LibreChat/assets/475367/e093a287-01fc-40b6-8b58-4c0bdb310d1c">

**Comment by Nihisil** [comment]
Ok, If I set `DALLE3_API_KEY` variable instead, it is working fine (while doc says that `DALLE_API_KEY` can be used for both versions)

**Reply by CXwudi** [reply]
Thanks for your debug, I just came across the same issue

**Reply by danny-avila** [reply]
@CXwudi PRs are welcome. Most of the documentation is written as a communal effort. I do take blame for this particular issue as the keys needed to be separated.

**Reply by CXwudi** [reply]
Honestly I would like to, but now I really confused on what is the purpose of `DALLE_API_KEY`?

Btw I deleted the blame

**Reply by danny-avila** [reply]
No worries. I meant to have it work for both but it requires a minor redesign of plug-in authentication. I would actually hold off and see if I  can find an easy way to make it work.

**Comment by danny-avila** [comment]
The docs are now correct in saying only `DALLE_API_KEY` is needed: https://github.com/danny-avila/LibreChat/pull/1760
